public/js/redux/chartsRedux.js

```diff
@@ -28,9 +30,15 @@ const chartsRedux = State({
         }
     },
 
-    updateComposerVsIncopy(state, composerVsInCopyData) {
+    updateComposerVsIncopy(state, { chartData, startDate, endDate }) {
+        const { composerResponse, inCopyResponse } = chartData;
+        const range = endDate.diff(startDate, 'days');
+        const composerData = composerResponse.data.length < range ?  fillMissingDates(startDate, endDate, composerResponse.data).sort(compareDates) : composerResponse.data;
```

Comment:
> But I think returning no data from the API when there were no articles started in a specific system and handling this client side makes sense.

I'm not sure I agree - the contract with the API describes the shape of the data and I think that returning with 'missing' datapoints is breaking that contract - the value of 'no stories published in incopy on x day' is `0` not `null` so the response should describe the data that way IMO.

i.e - this is a perfectly valid data point and should be a part of the response:

```json
 {
        "x" : 1502409600000,
        "y" : 0
    }
```

However, I think that as this is working client-side, it's probably not worth refactoring to the back-end at this point, especially if we want to make it more generic in the future.